### ISSUE TITLE
E0255 detected as E0252 and E0428

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8156]
## Environment

* **IntelliJ Rust plugin version:** 0.4.161.4262-213-nightly
* **Rust toolchain version:** 1.58.0-nightly (547a6ffee 2021-10-21) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3 RC (CL-213.5744.123)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
The code is incorrect in the indicated places, but the reported errors are false positive in some sense.
<img width="698" alt="изображение" src="https://user-images.githubusercontent.com/49211026/143018266-1a772ba7-ffd2-4af1-ba80-cfbbd7258293.png">

See [E0255](https://doc.rust-lang.org/error-index.html#E0255)

> You can’t import a value whose name is the same as another value defined in the module.

## Steps to reproduce
```rust
//A second item with name 'f' imported. Try to use an alias. [E0252]
use std::fmt as f;

// A type named `f` has already been defined in this module [E0428]
struct f {
    x: i8,
}

fn main() {}

```

**Comment by dima74** [comment]
Fixed by #9026
